src/order/base_order_utils.cairo

```diff
@@ -104,3 +105,48 @@ struct CreateOrderParams {
     /// The referral code linked to this order.
     referral_code: felt252
 }
+
+impl CreateOrderParamsClone of Clone<CreateOrderParams> {
+    fn clone(self: @CreateOrderParams) -> CreateOrderParams {
+        CreateOrderParams {
+            receiver: *self.receiver,
+            callback_contract: *self.callback_contract,
+            ui_fee_receiver: *self.ui_fee_receiver,
+            market: *self.market,
+            initial_collateral_token: *self.initial_collateral_token,
+            swap_path: self.swap_path.clone(),
+            size_delta_usd: *self.size_delta_usd,
+            initial_collateral_delta_amount: *self.initial_collateral_delta_amount,
+            trigger_price: *self.trigger_price,
+            acceptable_price: *self.acceptable_price,
+            execution_fee: *self.execution_fee,
+            callback_gas_limit: *self.callback_gas_limit,
+            min_output_amount: *self.min_output_amount,
+            order_type: *self.order_type,
+            decrease_position_swap_type: *self.decrease_position_swap_type,
+            is_long: *self.is_long,
+            should_unwrap_native_token: *self.should_unwrap_native_token,
+            referral_code: *self.referral_code
+        }
+    }
+}
+
+/// Checks if order is a market order.
+/// # Arguments
+/// * `order_type` - The order type.
+/// # Returns
+/// Whether the order is a market order.
+fn is_market_order(order_type: OrderType) -> bool {
+    order_type == OrderType::MarketSwap
+        || order_type == OrderType::MarketIncrease
+        || order_type == OrderType::MarketDecrease
+}
+
+/// Validates that an order exists.
+/// # Arguments
+/// * `order` - The order to check.
+fn validate_non_empty_order(order: Order) {
```

Comment:
this should take a `@Order`, not `Order` - and then you can pass it without doing an explicit clone

Reply:
understood - will update these